R/onLoad.R

```diff
@@ -19,13 +19,12 @@
   # Runs when loaded but not attached to search() path; e.g., when a package just Imports (not Depends on) data.table
   if (!exists("test.data.table", .GlobalEnv, inherits=FALSE)) {
     # check when installed package is loaded but skip when developing the package with cc()
-    dllV = if (is.loaded("CdllVersion",PACKAGE="datatable")) .Call(CdllVersion) else "before 1.12.0"
-    #                                              ^^ no dot as this is the name of the dll file, #3282
```

Comment:
@jangorecki what of the referenced issue?

> Local testing on Ubuntu shows there has to be no dot in the package name (that argument is really the name of the dll/so which has the dot removed).

Reply:
according to documentation it should be a dll name, so `data_table` now, but using that leads to failure linux
```
Error: package or namespace load failed for 'data.table':
 .onLoad failed in loadNamespace() for 'data.table', details:
  call: NULL
  error: invalid version specification 'before 1.12.0'
```
which is caused by the fact that it still uses `data.table/libs/data.table.so`.

Ultimately I think that R should raise and exception not allowing to create such dll because it is against WRE
> DLL needs to be both a valid file name and valid as part of a C entry point (e.g. it cannot contain ‘.’)

so looks like we need Makevars to rename dll